client/src/main/java/io/pravega/client/stream/impl/StreamSegments.java

```diff
@@ -103,10 +116,7 @@ private void verifyReplacementRange(StreamSegmentsWithPredecessors replacementSe
                                         replacementSegments);
             Preconditions.checkArgument(lowerReplacedSegment != null, "Missing replaced replacement segments %s",
                                         replacementSegments);
-            Preconditions.checkArgument(replacementRanges.containsKey(upperReplacedSegment.getValue().getSegmentId()),
-                                        "Inconsistant replaced replacement segments %s", replacementSegments);
-            Preconditions.checkArgument(replacementRanges.containsKey(lowerReplacedSegment.getValue().getSegmentId()),
-                                        "Inconsistant replaced replacement segments %s", replacementSegments);
+            // Note: replacedSegments can be already present in the current segment mapping in a scale down / merge scenario.
```

Comment:
The check is more complex, but we should still make some assertion...

Reply:
definition: 
- `allocated range: range assigned to segment by controller`
- `included range: range included for a given segment in the writer's map`
Now assertions/invariants here should be: 
1. union of all `allocated ranges` (and same for `included ranges`) should  [0.0, 1.0). 
2. No segment should have an `included range` that exceeds its `allocated range`
3. replacement segment's `included range` should 
 - be a subset of replacement segment's `allocated range`
 - overlap with `exiting` segment's `allocated range`